### PR TITLE
[32][35] Connect edit job page to API

### PR DESCRIPTION
**Pull request by PKhing**
## What did you do

- Connect edit job page to api
- Add missing job validation
- Fix edit job API
- Implement mock router (will write in notion)
- Add error text to date and time picker

## Demo

- login at https://dev.modela.miello.dev/login
username: `casting@gmail.com`
password: `password`
- go to https://dev.modela.miello.dev/job/13/edit and try to edit job

### CODE REVIEW

File: apps/web/src/modules/job/components/JobForm/hooks/useJobForm/index.ts [hunk 1]
```diff
@@ -17,17 +22,30 @@ const useJobForm = (defaultValues: IPostJobSchemaType, edit?: boolean) => {
...
-  const handleSuccess: SubmitHandler<IPostJobSchemaType> = useCallback(() => {
-    console.log('success')
-    if (edit) {
-      console.log('edit success')
-    }
-  }, [edit])
+  const handleSuccess: SubmitHandler<IPostJobSchemaType> = useCallback(
+    async (data) => {
+      const payload = fieldToPayload(data)
+      console.log('success')
```
Comment (miello) [inline apps/web/src/modules/job/components/JobForm/hooks/useJobForm/index.ts:36]: Consider remove `console.log`

File: apps/web/src/modules/job/components/JobForm/hooks/useJobForm/index.ts [hunk 2]
```diff
@@ -17,17 +22,30 @@ const useJobForm = (defaultValues: IPostJobSchemaType, edit?: boolean) => {
...
+      if (edit) {
+        const { jobId } = router.query as { jobId: string }
+        await apiClient.put<unknown, unknown, EditJobDto>(
+          `/job/${jobId}`,
+          payload,
+        )
```
Comment (miello) [inline apps/web/src/modules/job/components/JobForm/hooks/useJobForm/index.ts:42]: Do we need error handling ?

File: apps/web/src/common/utils/testing/mockUser/index.ts [hunk 3]
```diff
@@ -15,8 +18,8 @@ export const mockUser = () => {
   let returnUser: GetUserDto = {
     ...MOCK_USER,
     companyName: undefined,
-    isVerified: true,
-    type: UserType.ACTOR,
+    isVerified: isVerified,
+    type: type,
```
Comment (miello) [inline apps/web/src/common/utils/testing/mockUser/index.ts:22]:
```suggestion
    isVerified,
    type,
```

**Comment by miello** [comment]
image
image

Can we use 24hr format here ?

**Review by miello — CHANGES_REQUESTED** [review]

File: apps/web/src/modules/job/components/JobForm/hooks/useJobForm/index.ts (hunk 2 above)
Comment (PKhing) [reply apps/web/src/modules/job/components/JobForm/hooks/useJobForm/index.ts:42]: will do that in the next sprint :cat_smile:

**Review by miello — APPROVED** [review]

**Pull request merged**